src/Microsoft.Data.SqlClient/tests/ManualTests/ProviderAgnostic/ReaderTest/ReaderTest.cs

```diff
@@ -266,5 +267,118 @@ public static void TestMain()
+
+                using (DbCommand command = provider.CreateCommand())
+                {
+                    command.Connection = con;
+                    command.CommandText = sqlQueryOne;
+                    command.ExecuteNonQuery();
+                }
+                using (DbCommand command = provider.CreateCommand())
+                {
+                    command.Connection = con;
+                    command.CommandText = sqlQueryTwo;
+                    command.ExecuteNonQuery();
+                }
+
+                System.Data.SqlTypes.SqlGuid sqlguid = new System.Data.SqlTypes.SqlGuid(Guid.NewGuid());                 
+
+                using (SqlCommand sqlCommand = new SqlCommand("", con as SqlConnection))
+                {
+                    sqlCommand.CommandText = "INSERT INTO "+ tableNameOne +" VALUES (@CustomerId,@FirstName,@BoolCol,@ShortCol,@ByteCol,@LongCol)";
+                    sqlCommand.Parameters.AddWithValue(@"CustomerId",1);
+                    sqlCommand.Parameters.AddWithValue(@"FirstName", "Microsoft");
+                    sqlCommand.Parameters.AddWithValue(@"BoolCol", true);
+                    sqlCommand.Parameters.AddWithValue(@"ShortCol", 3274);
+                    sqlCommand.Parameters.AddWithValue(@"ByteCol", 253);
+                    sqlCommand.Parameters.AddWithValue(@"LongCol", 922222222222);
+                    sqlCommand.ExecuteNonQuery();
+                }
+                using (SqlCommand sqlCommand = new SqlCommand("", con as SqlConnection))
+                {
+                    sqlCommand.CommandText = "INSERT INTO "+tableNameTwo+" VALUES (@CustomerId,@DoubleCol,@SingleCol,@GUIDCol,@DateTimeCol,@DecimalCol,@DateTimeOffsetCol)";
+                    sqlCommand.Parameters.AddWithValue(@"CustomerId", 1);                  
+                    sqlCommand.Parameters.AddWithValue(@"DoubleCol", 10.7);
+                    sqlCommand.Parameters.AddWithValue(@"SingleCol", 123.546f);
+                    sqlCommand.Parameters.AddWithValue(@"GUIDCol", sqlguid);
+                    sqlCommand.Parameters.AddWithValue(@"DateTimeCol", dateTime);
+                    sqlCommand.Parameters.AddWithValue(@"DecimalCol", 280);
+                    sqlCommand.Parameters.AddWithValue(@"DateTimeOffsetCol", dtoffset);
+                    sqlCommand.ExecuteNonQuery();
+                }
+                using (SqlCommand sqlCommand = new SqlCommand("", con as SqlConnection))
+                {
+                    sqlCommand.CommandText = "select top 1 * from " + tableNameOne;
+                    DbDataReader reader = sqlCommand.ExecuteReader();
+
+                    reader.Read();
+
+                    Assert.True(reader[0].GetType() == typeof(int) && reader.GetFieldValue<int>(0) == 1);
```

Comment:
Don't use `Asset.True()` if it's possible. It won't report sufficient information about a failure. And don't mix them up.
For instance, in these cases, you can use `Assert.IsType<T>` and `Assert.Equal()`.

Reply:
reader[x].GetType() is returning System.RuntimeType, I added only Equal assertions.